### ISSUE TITLE
Working with spark-sql but not spark-submit and pyspark

### ISSUE THREAD

**Issue by morristm** [yaooqinn/spark-ranger#15]
I'm running HDP 3.1.5 it has ranger 1.2 and spark 2.3.  I configured spark-ranger per the instructions and it works with spark-sql, I can see that ranger is managing the access control to my hive tables and databases.  However if I run it with pyspark or spark-submit ranger is not controlling access control.   I am using this in yarn cluster mode.  I noticed spark-sql creates a local driver whereas pyspark and spark-submit create the driver on one of the yarn nodes.  Any ideas on what could be causing this?

**Comment by yaooqinn** [comment]
https://issues.apache.org/jira/browse/SPARK-25003
for spark-submit, please try to find any clues in am log

**Comment by morristm** [comment]
Thanks, when I see the issue with spark-submit I am also submitting a python job.  Is it possible that it also goes down the same code path?

From reading the comments in this spark issue, it looks like the patch is only available for Spark 3.0?  Is that how you understand it?

**Comment by yaooqinn** [comment]
yes, pyspark need that patch to be merged to the spark distribution you deploy
